src/OpenLoco/Company.cpp

```diff
@@ -469,4 +473,71 @@ namespace OpenLoco
         regs.ebx = enumValue(id());
         call(0x00437F47, regs);
     }
+
+    /* 0x004A6841
+     * Creates a vector of all the available rail track (trains and trams) for a company
+     * Tram track is marked with a (1<<7) flag within the uint8_t
+     */
+    std::vector<uint8_t> Company::getAvailableRailTracks()
+    {
+        std::vector<uint8_t> result;
+
+        std::unordered_set<uint8_t> tracks;
+        for (auto i = 0u; i < ObjectManager::getMaxObjects(ObjectType::vehicle); ++i)
```

Comment:
this loop is duplicated below and iterates over the same array of objects for `vehicle`, so the loops can be merged

Reply:
they insert to different vectors though and are looking for different things. I feel it looks better having different loops for it as there will never really be that many vehicle objects loaded.

Reply:
they are, but the overall process is the same, and more roughly half the loop body is duplicated. personally I would have combined them so save the double iteration, but yeah if `getMaxObjects(ObjectType::vehicle)` is low, then it isn't an issue